src/cpu/pred/btb/btb_mgsc.cc

```
@@ -109,6 +109,21 @@ BTBMGSC::initStorage()
     pWeightTable.resize(weightTableSize);
     biasWeightTable.resize(weightTableSize);
 
+    //perception weightTable init
+    auto allocWeightTable = [&](std::vector<std::vector<int16_t>> &table,unsigned numEntries) -> uint64_t {
+        assert(isPowerOf2(numEntries));
+        // if (!isPowerOf2(numEntries)) {
+        //     panic("numEntries (%d) is not a power of 2!", numEntries);
+        // }
+        table.resize(numEntries);
+        for (unsigned int i = 0; i<numEntries;++i){
```

Cppcheck (2.19.0)

[error] Reference to temporary returned.

(returnTempReference)